skipchain/api.go

```diff
@@ -177,10 +177,33 @@ func (c *Client) CreateRootControl(elRoot, elControl *onet.Roster,
 // 'latest' skipblock and the id (=hash) of the latest skipblock.
 func (c *Client) GetUpdateChain(roster *onet.Roster, latest SkipBlockID) (reply *GetUpdateChainReply, err error) {
 	reply = &GetUpdateChainReply{}
-	r := roster.RandomServerIdentity()
-	err = c.SendProtobuf(r,
-		&GetUpdateChain{latest}, reply)
-	return
+	for {
+		r2 := &GetUpdateChainReply{}
+		r := roster.RandomServerIdentity()
+		err = c.SendProtobuf(r, &GetUpdateChain{LatestID: latest}, r2)
+		if err != nil {
+			return
+		}
+		// Why would this happen?
+		if len(r2.Update) == 0 {
```

Comment:
If this shouldn't happen, then we need to at least log something or return an error?

Reply:
There may be malicious servers that can return this. So it can happen, and by Murphy's law, will happen. So the best thing here is to retry another server again as @ineiti suggested a few lines above.